### PR TITLE
Add env variable for node mapping validation to be loose to shortname, instead of fqdn

### LINKED ISSUE #465: PVC attachment error due to nodemapping issue in CNSA
**Describe the bug**
CNSA does maps openshift nodes to daemon node names but filesystem GET Rest API returns admin node names which causes the CSI validation for filesystem mount to fail, thus failing the PVC attach operation.

**Expected behavior**
PVC attach should succeed and should be mounted in pod

**Environment**
CNSA 5.1.1.3

**Additional context**
As a tactical solution filesystem mount validation should be skipped

### PR DESCRIPTION
**Pull request by dunnevan**
resolves #465

Opening a PR as a suggested interim workaround of issue #465.  Please add thoughts and considerations, this is mostly to get discussion going—

Suggested env variable name would be akin to existing SKIP_MOUNT_UNMOUNT.  This one would be called: SHORTNAME_NODE_MAPPING and value would either be "yes" or "no".  Default would be "no", so there is no behavior change.  This env variable need not be exposed or supported, beyond the limited workaround here.

### CODE REVIEW

**Comment by deeghuge** [comment]
Hi @dunnevan, Thanks for the PR. Changes looks good to me. 
I have one query : Are we sure that Admin short name which is used to create node mapping is always there in the filesystem mount node list(FQDN) reported by rest api ?

File: driver/csiplugin/gpfs_util.go [hunk 1]
```diff
@@ -335,3 +336,39 @@ func ConvertToBytes(inputStr string) (uint64, error) {
...
+const (
+	SHORTNAME_NODE_MAPPING = "SHORTNAME_NODE_MAPPING"
+	SKIP_MOUNT_UNMOUNT     = "SKIP_MOUNT_UNMOUNT"
+)
+
+func shortnameInSlice(shortname string, nodeNames []string) bool {
+	glog.V(6).Infof("gpfs_util shortnameInSlice. string: %s, slice: %v", shortname, nodeNames)
+	for _, name := range nodeNames {
+		short := strings.SplitN(name, ".", 2)[0]
+		if strings.EqualFold(short, shortname) {
```
Comment (smitaraut) [inline driver/csiplugin/gpfs_util.go:369]: I think it wont work in nodename examples like **compute-1** and **compute-10** ?

Comment (deeghuge) [reply driver/csiplugin/gpfs_util.go:369]: Hi Smita,  since equal fold does the exact comparison only without case it will work as we expect

**Comment by dunnevan** [comment]
> Hi @dunnevan, Thanks for the PR. Changes looks good to me.
> I have one query : Are we sure that Admin short name which is used to create node mapping is always there in the filesystem mount node list(FQDN) reported by rest api ?

No, I've heard that GUI does some of it's own name resolution.  If so this may not work.  It needs to be tested before being accepted.

With that said, it would work if REST API is returning as mmlscluster or mmlsmount would return

File: operator/roles/csiscaleoperator/defaults/main.yml [hunk 2]
```diff
@@ -56,3 +56,5 @@ state: present
 
 # Kubelet root directory path.
 kubeletRootDirPath: "{{ _csi_ibm_com_csiscaleoperator.spec.kubeletRootDirPath | default('/var/lib/kubelet') }}"
+
+shortnameNodeMapping: "{{ lookup('env', 'SHORTNAME_NODE_MAPPING') | default('no') }}"
```
Comment (deeghuge) [inline operator/roles/csiscaleoperator/defaults/main.yml:60]: for some reason default part is not working. when env is not specified in operator.yaml then remain empty in driver pod. I tried following version which I found working
_shortnameNodeMapping: "{{ lookup('env', 'SHORTNAME_NODE_MAPPING') | default('no', true) }}"_

**Review by deeghuge — APPROVED** [review]
lgtm

**Pull request merged**